rad/prelude/cmd-parsing.rad

```diff
@@ -27,32 +30,60 @@
   (fn [cmd]
     (string-append "Too many arguments for command \"" cmd "\"")))
 
-(def /cmd-1-opts
-  "Matches a command `cmd` with one argument and an optional
-  combination of options that are allowed by opts-map"
-  (fn [cmd opts opts-map]
+(def /cmd-opts
+  "Matches the command `cmd` and a combination of optional parameters, where
+  `opt-match` matches the options to the keys of `multi-ops`. `multi-ops` is a
+  dict of option-keys with multiple arguments. If no options are passed, the
+  `default` is returned."
+  (fn [cmd opts cmd-opts help]
+    (def default
+      (dict-from-seq
+        (map (fn [opt] (list (lookup :key opt) (lookup :default opt))) cmd-opts)))
+    (def-rec parse-cli
+      (fn [args]
+        (def parse-arg
+          (fn [opt args]
+            (def opt-key (lookup :key opt))
+            (match args
+              (/cons 'val 'rest)
+                  (if (elem? val (lookup :possible-values opt))
+                    (modify-map opt-key (fn [x] (cons val x)) (parse-cli rest))
```

Comment:
We should only do this if the type is `:multi-opt`. Otherwise we should just update the value.

Reply:
Do you mean you want to prepare the code for a possible third option where an option requires only a single value? I am not sure if it makes sense to include more option types here now without having explicit use for them, as there could be a variety of different option types.

Reply:
Good point, YAGNI. Let’s keep it like it is for now